pkg/apis/core/v1/utils.go

```diff
@@ -380,3 +395,661 @@ func (h *handler) checkBackupPointInUse(backups *v1.BackupList, name string) boo
+
+	return
+}
+
+func completeKubeadm(clientSet *kubernetes.Clientset, kubeadmConf *corev1.ConfigMap, c *v1.Cluster) error {
+	clusterData := kubeadmConf.Data["ClusterConfiguration"]
+	clusterConf := &v1.ClusterConfiguration{}
+	err := yaml.Unmarshal([]byte(clusterData), clusterConf)
+	if err != nil {
+		return err
+	}
+
+	proxyConf, err := clientSet.CoreV1().ConfigMaps("kube-system").
+		Get(context.TODO(), "kube-proxy", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	proxyMode := "iptables"
+	if strings.Contains(proxyConf.Data["config.conf"], "mode: ipvs") {
+		proxyMode = "ipvs"
+	}
+
+	c.Kind = "cluster"
+	c.APIVersion = "core.kubeclipper.io/v1"
+	c.Name = fmt.Sprintf("%s-%s", c.Name, clusterConf.ClusterName)
+
+	c.Provider.Name = v1.ClusterKubeadm
+	c.KubernetesVersion = clusterConf.KubernetesVersion
+	c.CertSANs = clusterConf.APIServer.CertSANs
+	c.LocalRegistry = clusterConf.ImageRepository
+	c.KubeProxy = v1.KubeProxy{}
+	c.Etcd.DataDir = clusterConf.Etcd.Local.DataDir
+	c.Kubelet.RootDir = ""
+	c.Networking.IPFamily = ""
+	serviceSubnet := strings.Split(clusterConf.Networking.ServiceSubnet, ",")
+	podSubnet := strings.Split(clusterConf.Networking.PodSubnet, ",")
+	if len(serviceSubnet) >= 2 && len(podSubnet) >= 2 {
+		c.Networking.IPFamily = v1.IPFamilyDualStack
+	} else {
+		c.Networking.IPFamily = v1.IPFamilyIPv4
+	}
+	c.Networking.Services = v1.NetworkRanges{CIDRBlocks: serviceSubnet}
+	c.Networking.Pods = v1.NetworkRanges{CIDRBlocks: podSubnet}
+	c.Networking.DNSDomain = clusterConf.Networking.DNSDomain
+	c.Networking.ProxyMode = proxyMode
+	c.Networking.WorkerNodeVip = ""
+	c.CNI = v1.CNI{}
+	c.Addons = []v1.Addon{}
+	c.Description = "kubeadm cluster import"
+
+	return nil
+}
+
+func (h *handler) clusterNodeCheck(c *v1.Cluster, ipMap map[string]string, ips []string) error {
+	nodeList, err := h.clusterOperator.ListNodes(context.TODO(), query.New())
+	if err != nil {
+		return err
+	}
+	for _, no := range nodeList.Items {
+		if _, ok := ipMap[no.Status.Ipv4DefaultIP]; ok {
+			return fmt.Errorf("node %s already exists", no.Status.Ipv4DefaultIP)
+		}
+		for _, addr := range no.Status.Addresses {
+			if _, ok := ipMap[addr.Address]; ok {
+				return fmt.Errorf("node %s already exists", no.Status.Ipv4DefaultIP)
+			}
+		}
+	}
+	if err := SudoCheck("sudo", c.Provider.SSHConfig.Convert(), ips); err != nil {
+		return err
+	}
+	// check if the node is already added
+	for _, nodeIP := range ips {
+		if !preCheckKcAgent(c, nodeIP) {
+			return fmt.Errorf("node %s kc-agent service already exists", nodeIP)
+		}
+	}
+
+	// TODO: Currently considering only direct connection networks
+	//return sudo.MultiNIC("ipDetect", o.deployConfig.SSHConfig, o.IOStreams, nodeIPs, c.ipDetect)
+	return nil
+}
+
+func SudoCheck(name string, sshConfig *sshutils.SSH, allNodes []string) error {
+	if sshConfig.User == "" {
+		return fmt.Errorf("current ssh user is empty")
+	}
+	// need enter passwd to run sudo
```

Comment:
The following code does the checking
	
if sshConfig.Password == "" && sshConfig.PkFile == "" && sshConfig.PkDataEncode == "" {
		return fmt.Errorf("The current user(%s) ssh pk-file/pk/paasword are empty", sshConfig.User)
	}

Reply:
> The following code does the checking
> 
> if sshConfig.Password == "" && sshConfig.PkFile == "" && sshConfig.PkDataEncode == "" { return fmt.Errorf("The current user(%s) ssh pk-file/pk/paasword are empty", sshConfig.User) }

for use sudo,we must got passed,instead of pkfile.